plugins/package-managers/node/src/main/kotlin/utils/NpmDetection.kt

```diff
@@ -17,237 +17,223 @@
+            }.flatten()
 
-    fun isYarnWorkspaceRoot(definitionFile: File) =
-        try {
-            definitionFile.readTree().has("workspaces")
-        } catch (e: JsonProcessingException) {
-            e.showStackTrace()
+            when {
+                manager !in managersFromWorkspaces -> return@filter false
+                managersFromWorkspaces.size == 1 -> {
+                    logger.info { "Skipping '$file' as it is part of an implicitly handled $manager workspace." }
+                    return@filter false
+                }
+            }
 
-            logger.error {
-                "Could not parse '${definitionFile.invariantSeparatorsPath}': ${e.collectMessages()}"
+            logger.warn {
+                val managers = managersFromFiles + managersFromWorkspaces
+                "Ignoring '$file' directory as the package manager is unclear. It could be any of " +
+                    "${managers.joinToString()}."
             }
 
             false
         }
-
-    val pnpmWorkspaceSubmodules = getPnpmWorkspaceSubmodules(definitionFiles)
-    val yarnWorkspaceSubmodules = getYarnWorkspaceSubmodules(definitionFiles)
-
-    return definitionFiles.map { definitionFile ->
-        PackageJsonInfo(
-            definitionFile = definitionFile,
-            isPnpmWorkspaceRoot = isPnpmWorkspaceRoot(definitionFile.parentFile),
-            isYarnWorkspaceRoot = isYarnWorkspaceRoot(definitionFile) &&
-                !isPnpmWorkspaceRoot(definitionFile.parentFile),
-            hasYarnLockfile = hasYarnLockFile(definitionFile.parentFile),
-            hasNpmLockfile = hasNpmLockFile(definitionFile.parentFile),
-            hasPnpmLockfile = hasPnpmLockFile(definitionFile.parentFile),
-            hasYarn2ResourceFile = hasYarn2ResourceFile(definitionFile.parentFile),
-            isPnpmWorkspaceSubmodule = definitionFile in pnpmWorkspaceSubmodules,
-            isYarnWorkspaceSubmodule = definitionFile in yarnWorkspaceSubmodules &&
-                definitionFile !in pnpmWorkspaceSubmodules
-        )
-    }
 }
 
-private fun getPnpmWorkspaceMatchers(definitionFile: File): List<PathMatcher> {
-    @JsonIgnoreProperties(ignoreUnknown = true)
-    data class PnpmWorkspaces(val packages: List<String> = emptyList())
-
-    fun String.isComment() = trim().startsWith("#")
-
-    val pnpmWorkspaceFile = definitionFile.resolveSibling("pnpm-workspace.yaml")
-
-    return if (pnpmWorkspaceFile.isFile && pnpmWorkspaceFile.readLines().any { !it.isComment() }) {
-        val workspaceMatchers = pnpmWorkspaceFile.readValue<PnpmWorkspaces>().packages
-
-        workspaceMatchers.map { matcher ->
-            val pattern = "glob:${definitionFile.parentFile.invariantSeparatorsPath}/${matcher.removeSuffix("/")}"
-            FileSystems.getDefault().getPathMatcher(pattern)
+/**
+ * An enum of all supported Node package managers.
+ */
+enum class NodePackageManager(
+    val lockFileName: String,
+    val otherFileName: String? = null,
```

Comment:
I had `markerFileName` in between, but then went for something "looser" as I wasn't yet sue about all of its uses, but I agree it could now be `markerFileName` or `indicatorFileName` again. Which would you prefer?